woocommerce/woocommerce-ios
# [Mobile payments] Add all mandatory fields to receipt

Closes #3978 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader ⚠️

We continue building our prototype (for reference, p91TBi-4q4-p2), this time, by continuing the work started in #3997, with a first iteration on the layout of the printed receipt. This PR begins where #4015 ends.

https://user-images.githubusercontent.com/2722505/115871990-42603c80-a40f-11eb-83a1-2f4e37c29ace.mov

The receipt generated, in pdf format: 
[Order Receipt 2.pdf](https://github.com/woocommerce/woocommerce-ios/files/6365916/Order.Receipt.2.pdf)

Custom receipts must contain certain information:
> You can also customize receipts to include any design and content you want—as long as you list required information. When you accept in-person payments with EMV chip cards, card networks require you to include several fields on the receipts you provide to customers.
[Source](https://stripe.com/docs/terminal/checkout/receipts#custom)

So this PR renders that mandatory information into the footer of the receipt. And since we are at it, it also sets the foundation to print the receipt by declaring its content as HTML. Why? Because the simplest way to render a receipt, with a certain degree of customization, is by printing HTML rendered via a `UIMarkupTextPrintFormatter`. 

## Changes
* Update existing model objects to pass to the receipt some extra metadata, like the store name. We will be passing this metadata in the PaymentIntent metadata property. This is something I have done based on my gut feeling, not because the documentation recommends to do so
* Update `ReceiptRenderer` to render a receipt using a UIMarkupTextPrintFormatter. This is something that might be worth pursuing because a) it's simple and b) in theory, we could extract the code that generates and renders the HTML to a separate entity that could helps us with #3982 and, in the long term, with generating receipts locally (#3975)
* Some other bits here and there (user-facing card brands, price of receipt line items)


## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* At that point, tapping the "Print receipt" button will trigger a printout. The "Email receipt button" does nothing yet, and the "Do nothing button will dismiss the modal view.
* Notice footer with the mandatory fields for payments collected in the US as [per Stripe's documentation](https://stripe.com/docs/terminal/checkout/receipts#custom)

Update release notes:
